src/typeguard.ts

```diff
@@ -2,53 +2,6 @@
 // Author: Ron B. Yeh
 // MIT License
 
-import { Dot } from './dot';
-import { GraceNote } from './gracenote';
-import { GraceNoteGroup } from './gracenotegroup';
-import { Note } from './note';
-import { Barline } from './stavebarline';
-import { StaveNote } from './stavenote';
-import { StemmableNote } from './stemmablenote';
-import { TabNote } from './tabnote';
-import { globalObject } from './util';
-
-// eslint-disable-next-line
-export function isHTMLCanvas(element: any): element is HTMLCanvasElement {
-  if (!element) return false;
-
-  const global = globalObject();
-  return (
-    // It's either an instance of the HTMLCanvasElement class,
-    (typeof global.HTMLCanvasElement === 'function' && element instanceof global.HTMLCanvasElement) ||
-    // OR it's pretending to be a <canvas> element. Good enough!
-    // Do not rely on .tagName, because node-canvas doesn't provide a tagName.
-    (typeof element.getContext === 'function' && typeof element.toDataURL === 'function')
-  );
-}
-
-// eslint-disable-next-line
-export function isHTMLDiv(element: any): element is HTMLDivElement {
-  if (!element) return false;
-
-  const global = globalObject();
-  return (
-    // It's either an instance of the HTMLDivElement class.
-    (typeof global.HTMLDivElement === 'function' && element instanceof global.HTMLDivElement) ||
-    // OR it's pretending to be a <div>. See: svgcontext.ts.
-    (typeof element.appendChild === 'function' && typeof element.style === 'object')
-  );
-}
-
-// Helper functions for checking an object's type, via `instanceof` and `obj.constructor.CATEGORY`.
-export const isNote = (obj: unknown): obj is Note => isCategory(obj, Note);
-export const isStemmableNote = (obj: unknown): obj is StemmableNote => isCategory(obj, StemmableNote);
-export const isStaveNote = (obj: unknown): obj is StaveNote => isCategory(obj, StaveNote);
-export const isTabNote = (obj: unknown): obj is TabNote => isCategory(obj, TabNote);
-export const isGraceNote = (obj: unknown): obj is GraceNote => isCategory(obj, GraceNote);
-export const isGraceNoteGroup = (obj: unknown): obj is GraceNoteGroup => isCategory(obj, GraceNoteGroup);
-export const isDot = (obj: unknown): obj is Dot => isCategory(obj, Dot);
-export const isBarline = (obj: unknown): obj is Barline => isCategory(obj, Barline);
```

Comment:
So, why are we losing these?

Reply:
To avoid the circular dependencies

Reply:
I mentioned this up thread, haha. Yes this PR is too long. 

https://github.com/0xfe/vexflow/pull/1232#discussion_r765603020

There typeguard methods started breaking the build so I moved each typeguard out to its related class. 

So now, isDot() is in the Dot class file. This breaks up the circular dependencies because typeguard.ts doesn't have to import all these Element subclasses.

Reply:
Okay, got it -- I prefer to move these calls closer to the implementations anyway.